File: app/src/main/java/com/boji/withindexlistview/MainActivity.java
package com.boji.withindexlistview;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import com.boji.listview.WithIndexExpandableListView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private WithIndexExpandableListView listView ;
    private MyAdapter adapter = new MyAdapter();
    private ArrayList<String> charactes ;
    private List<ArrayList<String>> data ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();
        initData() ;
    }

    private void initView() {
        listView = findViewById(R.id.indexer);
    }

    private void initData() {
        charactes = new ArrayList<>(26) ;
        data = new ArrayList<>() ;
        char c = 'A' ;
        for (int i=c;i<='Z';i++) {
            ArrayList<String> contact = new ArrayList<>(10) ;
            for (int j=0;j<10;j++) {
                contact.add(String.valueOf((char)i) + (j + 1)) ;
            }
            data.add(contact) ;
            charactes.add(String.valueOf((char)i)) ;
        }

        adapter.setData(charactes,data);
        listView.setWithIndexAdapter(adapter);
    }
}

File: app/src/main/java/com/boiqin/MainActivity.kt
package com.boiqin

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setListener()
    }

    private fun setListener() {
        findViewById<Button>(R.id.btn_lv).setOnClickListener {
            val intent = Intent()
            intent.setClass(this, ListViewActivity::class.java)
            startActivity(intent)
        }
        findViewById<Button>(R.id.btn_elv).setOnClickListener {
            val intent = Intent()
            intent.setClass(this, ExpandableListViewActivity::class.java)
            startActivity(intent)
        }
    }
}
